src/platform/test/functional/services/dashboard/add_panel.ts

```diff
@@ -14,62 +14,68 @@ export class DashboardAddPanelService extends FtrService {
   private readonly header = this.ctx.getPageObject('header');
   private readonly savedObjectsFinder = this.ctx.getService('savedObjectsFinder');
-  private readonly browser = this.ctx.getService('browser');
   private readonly toasts = this.ctx.getService('toasts');
 
-  async clickOpenAddPanel() {
-    this.log.debug('DashboardAddPanel.clickOpenAddPanel');
+  async clickTopNavAddMenu() {
+    this.log.debug('DashboardAddPanel.clickTopNavAddMenu');
+    await this.testSubjects.click('dashboardAddTopNavButton');
+  }
+
+  async clickAddFromLibrary() {
+    this.log.debug('DashboardAddPanel.clickAddFromLibrary');
+    await this.clickTopNavAddMenu();
     await this.testSubjects.click('dashboardAddFromLibraryButton');
-    // Give some time for the animation to complete
-    await this.common.sleep(500);
+    await this.testSubjects.existOrFail('savedObjectsFinderTable');
+    await this.savedObjectsFinder.waitForListLoading();
   }
 
   async clickCreateNewLink() {
     this.log.debug('DashboardAddPanel.clickAddNewPanelButton');
-    await this.retry.try(async () => {
-      // prevent query bar auto suggest from blocking button
-      await this.browser.pressKeys(this.browser.keys.ESCAPE);
-      await this.testSubjects.click('dashboardAddNewPanelButton');
-      await this.testSubjects.waitForDeleted('dashboardAddNewPanelButton');
-      await this.header.waitUntilLoadingHasFinished();
-      await this.testSubjects.existOrFail('lnsApp', {
-        timeout: 5000,
-      });
+    await this.clickTopNavAddMenu();
+    await this.testSubjects.click('dashboardCreateNewVisButton');
+    await this.testSubjects.waitForDeleted('dashboardCreateNewVisButton');
+    await this.header.waitUntilLoadingHasFinished();
+    await this.testSubjects.existOrFail('lnsApp', {
+      timeout: 5000,
     });
   }
 
   async clickAddCustomVisualization() {
-    await this.clickEditorMenuButton();
+    await this.openAddPanelFlyout();
     await this.clickAddNewPanelFromUIActionLink('Custom visualization');
   }
   async clickAddMarkdownPanel() {
-    await this.clickEditorMenuButton();
+    this.log.debug('DashboardAddPanel.clickAddMarkdownPanel');
+    await this.openAddPanelFlyout();
     await this.clickAddNewPanelFromUIActionLink('Markdown text');
   }
 
   async clickAddMapPanel() {
-    await this.clickEditorMenuButton();
+    this.log.debug('DashboardAddPanel.clickAddMapPanel');
+    await this.openAddPanelFlyout();
     await this.clickAddNewPanelFromUIActionLink('Maps');
   }
 
   async clickAddLensPanel() {
-    await this.clickEditorMenuButton();
+    this.log.debug('DashboardAddPanel.clickAddLensPanel');
+    await this.openAddPanelFlyout();
     await this.clickAddNewPanelFromUIActionLink('Lens');
   }
 
-  async clickEditorMenuButton() {
-    this.log.debug('DashboardAddPanel.clickEditorMenuButton');
-    await this.testSubjects.click('dashboardEditorMenuButton');
+  async openAddPanelFlyout() {
```

Comment:
Yes, `clickEditorMenuButton` was the worst name. `openAddPanelFlyout` is so much better!!!